src/Symfony/Component/VarExporter/Instantiator.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarExporter;
+
+use Symfony\Component\VarExporter\Internal\Hydrator;
+use Symfony\Component\VarExporter\Internal\Registry;
+
+/**
+ * A utility class to create objects without calling their constructor.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class Instantiator
+{
+    /**
+     * Creates an object and sets its properties without calling its constructor nor any other methods.
+     *
+     * For example:
+     *
+     *     // creates an empty instance of Foo
+     *     Instantiator::instantiate([Foo::class => []]);
+     *
+     *     // creates a Foo instance and sets one of its public, protected or private properties
+     *     Instantiator::instantiate([Foo::class => ['propertyName' => $propertyValue]]);
+     *
+     *     // creates a Foo instance and sets a private property defined on its parent Bar class
+     *     Instantiator::instantiate([
+     *         Foo::class => [],
+     *         Bar::class => ['privatePropertyName' => $propertyValue],
+     *     ]);
+     *
+     * Instances of ArrayObject, ArrayIterator and SplObjectHash can be created
+     * by using the special "\0" property name to define their internal value:
+     *
+     *     // creates an SplObjectHash where $info1 is attached to $obj1, etc.
+     *     Instantiator::instantiate([SplObjectStorage::class => ["\0" => [$obj1, $info1, $obj2, $info2...]]]);
+     *
+     *     // creates an ArrayObject populated with $inputArray
+     *     Instantiator::instantiate([ArrayObject::class => ["\0" => $inputArray]]);
+     *
+     * @param array $propertiesByClass The properties to set on the object, keyed by the class to bind to
```

Comment:
Please document the type for keys and values here: should probably be `array<string, array<string, mixed>>` from the examples above, but would suggest extracting the first key

Reply:
What's the syntax to document this, one that would be supported by phpstorm and phpdocumentor? For now, I described the expected structure in the param description and in the example. That could be enough in practice (`array<string, array<string, mixed>>` is a bit too abstract to be useful I fear)